opnode/rollup/sync/start_test.go

```diff
@@ -90,224 +99,217 @@ type syncStartTestCase struct {
-			ActualL1:          "abcde",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'e',
-			ExpectedRefL2:     'D',
-			ExpectedErr:       nil,
+			Name:               "already synced",
+			OffsetL2:           0,
+			EngineL1:           "abcde",
+			EngineL2:           "ABCDE",
+			ActualL1:           "abcde",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "",
+			ExpectedRefL2:      'E',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "genesis",
-			OffsetL2:          0,
-			EngineL1:          "a",
-			EngineL2:          "A",
-			ActualL1:          "a",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'a',
-			ExpectedRefL2:     0, // actual zero hash before genesis hash
-			ExpectedErr:       nil,
+			Name:               "genesis",
+			OffsetL2:           0,
+			EngineL1:           "a",
+			EngineL2:           "A",
+			ActualL1:           "a",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "",
+			ExpectedRefL2:      'A',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "reorg two steps back",
-			OffsetL2:          0,
-			EngineL1:          "abc",
-			EngineL2:          "ABC",
-			ActualL1:          "axy",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'x',
-			ExpectedRefL2:     'A',
-			ExpectedErr:       nil,
+			Name:               "reorg two steps back",
+			OffsetL2:           0,
+			EngineL1:           "abc",
+			EngineL2:           "ABC",
+			ActualL1:           "axy",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "xy",
+			ExpectedRefL2:      'A',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "Orphan block",
-			OffsetL2:          0,
-			EngineL1:          "abcd",
-			EngineL2:          "ABCD",
-			ActualL1:          "abcx",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'x',
-			ExpectedRefL2:     'C',
-			ExpectedErr:       nil,
+			Name:               "Orphan block",
+			OffsetL2:           0,
+			EngineL1:           "abcd",
+			EngineL2:           "ABCD",
+			ActualL1:           "abcx",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "x",
+			ExpectedRefL2:      'C',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "L2 chain ahead",
-			OffsetL2:          0,
-			EngineL1:          "abcdef",
-			EngineL2:          "ABCDEF",
-			ActualL1:          "abc",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 0,
-			ExpectedRefL2:     0,
-			ExpectedErr:       ethereum.NotFound,
+			Name:               "L2 chain ahead",
+			OffsetL2:           0,
+			EngineL1:           "abcdef",
+			EngineL2:           "ABCDEF",
+			ActualL1:           "abc",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "",
+			ExpectedRefL2:      'C',
+			ExpectedErr:        nil,
```

Comment:
@protolambda. I changed the error here. This now walks back until it finds the first L2 that has a matching L1. I guess this is more an L1 rollback instead of an L1 re-org.

Reply:
Hmm, was the `NotFound` error correctly announcing this case? I like the idea of changing the name to `FindSafeL2Head` to indicate that it might roll back. The re-org case is a little similar to the get ahead case.

I guess what would be the usage & behavior of the function when L2 is ahead of L1?